src/main/ngapp/accio/src/app/mask.directive.ts

```diff
@@ -0,0 +1,142 @@
+import { Directive, HostListener, ElementRef, Input } from '@angular/core';
+import { MagicWandService } from './magic-wand.service';
+
+@Directive({
+  selector: '[appMask]',
+})
+export class MaskDirective {
+  @Input() scale: number;
+  @Input() originalData: ImageData;
+  @Input() scaledData: ImageData;
+  @Input() maskData: ImageData;
+
+  // create direct reference of canvas on editor.html
+  constructor(
+    private canvas: ElementRef<HTMLCanvasElement>, 
+    private magicWandService: MagicWandService,
+  ) { }
+  private ctx: CanvasRenderingContext2D;
+  private maskImage: HTMLImageElement;
+
+  @HostListener('click', ['$event'])
+  onClick(e: MouseEvent) {
+    const xCoord = e.offsetX;
+    const yCoord = e.offsetY;
+    this.ctx = this.canvas.nativeElement.getContext('2d');
+
+    // tolerance will come from user input
+    const tolerance = 20;
+
+    //  TODO-MASK-maskPixels needs to contain cumulative set of all pixels user clicked and added to the mask
+    // returns an array indices of each pixel in the mask
+    const maskPixels = this.magicWandService.floodfill(this.originalData, Math.floor(xCoord / this.scale), Math.floor(yCoord / this.scale), tolerance);
```

Comment:
Consider having a function that will convert between canvas coordinates and image coordinates.  This way you can implement it once and use it every where.  Additionally, if you need to update it to handle panning, etc. you can easily find all the places you need to update your code to handle this.